doc/howtos/software-update-server.rst

```diff
@@ -0,0 +1,279 @@
+
+Bundles definition
+==================
+
+The documentation for using SW Update in Yocto Project can
+be found in the `swupd layer`_ and is only summarized here.
+These instructions assume that the work will happen on top of an
+image flavor that has swupd enabled, for example by building an image,
+such as with::
+
+   $ bitbake ostro-image-swupd
+
+or by downloading a pre-compiled image with ``-swupd`` in its name
+from the `Ostro OS download server`_
+
+.. _`Ostro OS download server`: https://download.ostroproject.org
+
+These are the main steps:
+
+- In ``build/conf/local.conf`` or equivalent location, set the
+  `OS_VERSION` variable to an integer value.
+  This number must be increased before each build that generates
+  swupd update artefacts.
+
+- Assign a list of bundle names to ``SWUPD_BUNDLES``::
+
+    SWUPD_BUNDLES = "feature_one feature_two"
+
+- For each named bundle, setup a varflag of ``BUNDLE_CONTENTS``
+  that matches the bundle name, and initialize it with a list of
+  packages whose content must be included in the bundle::
+
+    BUNDLE_CONTENTS[feature_one] = "package_one package_three package_six"
+
+Creating a Custom Image and matching Software Update Repository
+===============================================================
+
+Some of the steps listed in this section are optional and cover
+advanced configuration functionality.
+Not all the users will need them, if the default configuration of
+the Ostro OS meets their needs.
+
+The process is exemplified by using real components and describing
+how each step looks like, when applied to such components.
+Unless stated otherwise, the changes described are to be intended
+for the file ``build/conf/local.conf``
+
+
+Step 0: Review and optionally modify the content of the os-core bundle
+----------------------------------------------------------------------
+The os-core bundle comes with a predefined set of components,
+however each device is likely to have slighlty differnt requirements.
+In some cases it might be necessary to adjust the set of components
+included in os-core.
+
+To do this, either modify or extend the `OSTRO_IMAGE_FEATURES_REFERENCE`
+and the `OSTRO_IMAGE_INSTALL_REFERENCE` variables for changes
+respectively in features and components.
+
+Why modifying os-core? Because it's the only unremovable bundle and
+therefore it is well suited to deliver tho the device all those tools
+that must always be available.
+
+Example::
+
+  ``OSTRO_IMAGE_INSTALL_REFERENCE += " sudo"``
+
+Will add the content of the ``sudo`` recipe to the bundle os-core.
+
+.. note::
+   The step shown works only for adding components to os-core.
+
+   This is the typical case for customizing Ostro Os, because the
+   os-core configuration is very minimal.
+
+   However, in case one needs to remove something from os-core,
+   it is necessary to either create a new image recipe or a .bbappend
+   file, where to modify the variables that define the content
+   of os-core.
+
+   Example::
```

Comment:
I don't understand the example, is this a list of variables that should be modified in the recipe/.bbappend to modify the content of os-core? If so, can you make this clearer and perhaps list them out in separate lines.